source/agora/consensus/protocol/Nominator.d

```diff
@@ -213,6 +213,39 @@ extern(D):
         }
     }
 
+    /***************************************************************************
+
+        Create Stellar SCP object
+
+        A validator creates SCP object with the UTXO for enrollment. This
+        checks that the UTXO key for enrollment exists and new SCP should be
+        created because of new enrollment of this node.
+
+    ***************************************************************************/
+
+    public void recreateSCPObject () nothrow @safe
+    {
+        try
+        {
+            auto self_enroll = this.enroll_man.getEnrollmentKey();
+            if (self_enroll != this.scp_utxo)
+            {
```

Comment:
```suggestion
            // Either there's already an SCP object, or we're not enrolled
            if (self_enroll == this.scp_utxo)
                return;
```

In general, it's easier to get special cases "out of the way" and use early return if you can.
Reducing the level of indentation tends to reduce cognitive load.
And I think the special case of `Hash.init` should be documented... Which leads to another question: Can this method be called in a situation where `getEnrollmentKey` would return `Hash.init` where it was previously returning a non `init` value ? Because then we would (wrongly) create a new SCP object, correct ?

Reply:
It's a rare case that new enrollment is used for creating an SCP object. So I will just check it is `Hash.init` or not by removing the new `scp_utxo`. I will add a comment that we should check the situation where new enrollment is used for SCP env.